blocktopmenu.php

```diff
@@ -1358,4 +1371,104 @@ public function renderList()
 
         return $helper->generateList($links, $fields_list);
     }
+
+    private function createCategoryNode(array $category, $addThumbnails = false)
+    {
+        if (!$category['active']) {
```

Comment:
I don't want to divert from the objective of the pull-request (which is very sane), but since it change prototype, I thought there is room for improvements impossible before.
Use case for inactive categories:
In our shop, we use to menus using two distinct classification (there are two path to reach a product).
Thus, blocktopmenu displays level 1 categories, except inactive one.
Then we use a hook (which trigger the HTML-generator) for the inactive level 1 categorie which contains all the subcategories we want to display as our "second menu".
The default blocktopmenu behavior (not display inactive categories) is right, but the function is useful for other uses.

More explicitely:
```php
    // in a hook
    $categories = Category::getNestedCategories(39, 1, false, 1);
    $our_second_menu = $btm->generateCategoriesMenu($categories[39]['children'], 0, TRUE);
    $this->smarty->assign('OUR_SECOND_MENU', $our_second_menu);
```
For this to work, the generator should:
1. be public
2. accept a boolean parameter in order to ignore/accept inactive categories